zmon-security-common/src/main/java/org/zalando/zmon/config/ZmonOAuth2Properties.java

```diff
@@ -28,9 +43,30 @@
 
 	private boolean platformEnabled;
 
-	public String getClientId() {
-		return clientId;
-	}
+    @PostConstruct
+    public void init() {
+        additionalParams = UriComponentsBuilder.fromHttpUrl(authorizeUrl)
+                .build()
+                .getQueryParams()
+                .toSingleValueMap();
+
+        /**
+         * Following URLs are being used by "spring-social-zauth" library which expects only base URLs without
+         * additional query parameters. Thus, all possible query parameters are stripped.
+         *
+         * Problem of "spring-social-zauth" is that it appends "?client-id=" to the URL. Thus, if one provides an URL
+         * like "https://foo.bar/baz?param1=bro", the library will transform it to
+         * "https://foo.bar/baz?param1=bro?client-id=...".
```

Comment:
@jbellmann please take a look at this comment. Need your suggestion about contributing to "spring-social-zauth"